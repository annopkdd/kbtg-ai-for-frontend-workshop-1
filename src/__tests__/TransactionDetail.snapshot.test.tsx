import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router';
import TransactionDetail from '../pages/TransactionDetail';

// Mock the transaction data for consistent snapshots
jest.mock('../data/transactions', () => ({
  getAllTransactions: jest.fn(() => [
    {
      id: 'TXN001',
      type: 'outgoing',
      recipient: 'สมชาย ใจดี',
      payTag: '@somchai123',
      amount: 500,
      memo: 'ค่าอาหารเที่ยง',
      date: 'วันนี้ 14:30',
      timestamp: new Date('2025-06-19T14:30:00'),
      status: 'completed',
      category: 'food',
      reference: 'REF001234567'
    },
    {
      id: 'TXN002',
      type: 'incoming',
      sender: 'นิดา สวยงาม',
      payTag: '@nida456',
      amount: 1200,
      memo: 'คืนเงินค่าหนัง',
      date: 'เมื่อวาน 19:45',
      timestamp: new Date('2025-06-18T19:45:00'),
      status: 'completed',
      category: 'entertainment',
      reference: 'REF001234568'
    },
    {
      id: 'TXN003',
      type: 'outgoing',
      recipient: 'ประยุทธ์ มั่นคง',
      payTag: '@prayuth789',
      amount: 2500,
      memo: 'ค่าเช่าบ้าน',
      date: '2 วันที่แล้ว',
      timestamp: new Date('2025-06-17T10:00:00'),
      status: 'pending',
      category: 'bills',
      reference: 'REF001234569'
    }
  ])
}));

// Mock useParams to provide transaction ID
const mockUseParams = jest.fn();
jest.mock('react-router', () => ({
  ...jest.requireActual('react-router'),
  useParams: () => mockUseParams(),
  Link: ({ children, to, ...props }: any) => <a href={to} {...props}>{children}</a>
}));

describe('TransactionDetail Page Snapshot Tests', () => {
  beforeEach(() => {
    jest.useFakeTimers();
    jest.setSystemTime(new Date('2025-06-19T10:00:00Z'));
  });

  afterEach(() => {
    jest.useRealTimers();
    jest.clearAllMocks();
  });

  test('TransactionDetail page with valid outgoing transaction should match snapshot', () => {
    mockUseParams.mockReturnValue({ id: 'TXN001' });
    
    const { container } = render(
      <MemoryRouter>
        <TransactionDetail />
      </MemoryRouter>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test('TransactionDetail page with valid incoming transaction should match snapshot', () => {
    mockUseParams.mockReturnValue({ id: 'TXN002' });
    
    const { container } = render(
      <MemoryRouter>
        <TransactionDetail />
      </MemoryRouter>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test('TransactionDetail page with pending transaction should match snapshot', () => {
    mockUseParams.mockReturnValue({ id: 'TXN003' });
    
    const { container } = render(
      <MemoryRouter>
        <TransactionDetail />
      </MemoryRouter>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test('TransactionDetail page with invalid transaction ID should match snapshot', () => {
    mockUseParams.mockReturnValue({ id: 'INVALID_ID' });
    
    const { container } = render(
      <MemoryRouter>
        <TransactionDetail />
      </MemoryRouter>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test('TransactionDetail page with receipt shown should match snapshot', () => {
    mockUseParams.mockReturnValue({ id: 'TXN001' });
    
    const { container } = render(
      <MemoryRouter>
        <TransactionDetail />
      </MemoryRouter>
    );
    
    // Click show receipt button
    const receiptButton = container.querySelector('button[type="button"]');
    if (receiptButton && receiptButton.textContent?.includes('แสดงใบเสร็จ')) {
      (receiptButton as HTMLElement).click();
    }
    
    expect(container.firstChild).toMatchSnapshot();
  });
});