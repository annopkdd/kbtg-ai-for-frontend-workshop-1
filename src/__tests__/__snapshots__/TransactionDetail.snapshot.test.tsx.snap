// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`TransactionDetail Page Snapshot Tests TransactionDetail page with invalid transaction ID should match snapshot 1`] = `
<div
  class="min-h-screen bg-neutral-50 flex items-center justify-center"
>
  <div
    class="text-center"
  >
    <h1
      class="text-2xl font-bold text-neutral-900 mb-4"
    >
      ไม่พบรายการธุรกรรม
    </h1>
    <a
      class="bg-web-green-500 hover:bg-web-green-600 text-white px-6 py-3 rounded-lg transition-colors"
      href="/home"
    >
      กลับหน้าหลัก
    </a>
  </div>
</div>
`;

exports[`TransactionDetail Page Snapshot Tests TransactionDetail page with pending transaction should match snapshot 1`] = `
<div
  class="min-h-screen bg-neutral-50"
>
  <div
    class="bg-gradient-to-r from-web-green-600 to-web-green-500 text-white p-4"
  >
    <div
      class="flex items-center space-x-3 mb-4"
    >
      <a
        class="p-2 hover:bg-white/20 rounded-lg transition-colors"
        href="/home"
      >
        <svg
          class="w-5 h-5"
          fill="none"
          stroke="currentColor"
          viewBox="0 0 24 24"
        >
          <path
            d="M15 19l-7-7 7-7"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
      </a>
      <h1
        class="text-xl font-semibold"
      >
        รายละเอียดธุรกรรม
      </h1>
    </div>
  </div>
  <div
    class="p-4 space-y-4"
  >
    <div
      class="bg-white rounded-lg p-6 shadow"
    >
      <div
        class="text-center mb-6"
      >
        <div
          class="w-16 h-16 rounded-full flex items-center justify-center mx-auto mb-4 bg-warning/10"
        >
          <svg
            class="w-8 h-8 text-warning"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
        </div>
        <p
          class="text-3xl font-bold mb-2 text-error"
        >
          -
          ฿
          2,500
        </p>
        <p
          class="text-lg font-medium text-neutral-900 mb-1"
        >
          โอนเงินให้
           
          ประยุทธ์ มั่นคง
        </p>
        <p
          class="text-sm text-neutral-600"
        >
          @prayuth789
        </p>
        <div
          class="inline-flex items-center px-3 py-1 rounded-full text-sm font-medium mt-3 bg-warning/10 text-warning"
        >
          รอดำเนินการ
        </div>
      </div>
    </div>
    <div
      class="bg-white rounded-lg p-6 shadow"
    >
      <h3
        class="text-lg font-semibold text-neutral-900 mb-4"
      >
        รายละเอียด
      </h3>
      <div
        class="space-y-4"
      >
        <div
          class="flex justify-between"
        >
          <span
            class="text-neutral-600"
          >
            หมายเลขอ้างอิง
          </span>
          <span
            class="text-neutral-900 font-medium"
          >
            REF001234569
          </span>
        </div>
        <div
          class="flex justify-between"
        >
          <span
            class="text-neutral-600"
          >
            วันที่และเวลา
          </span>
          <span
            class="text-neutral-900 font-medium"
          >
            17 มิถุนายน 2568 เวลา 10:00:00
          </span>
        </div>
        <div
          class="flex justify-between"
        >
          <span
            class="text-neutral-600"
          >
            บันทึก
          </span>
          <span
            class="text-neutral-900 font-medium"
          >
            ค่าเช่าบ้าน
          </span>
        </div>
        <div
          class="flex justify-between"
        >
          <span
            class="text-neutral-600"
          >
            จำนวนเงิน
          </span>
          <span
            class="text-neutral-900 font-medium"
          >
            ฿
            2,500
          </span>
        </div>
        <div
          class="flex justify-between"
        >
          <span
            class="text-neutral-600"
          >
            ค่าธรรมเนียม
          </span>
          <span
            class="font-medium text-green-600"
          >
            ฟรี
          </span>
        </div>
        <div
          class="flex justify-between"
        >
          <span
            class="text-neutral-600"
          >
            หมวดหมู่
          </span>
          <span
            class="text-neutral-900 font-medium"
          >
            ค่าใช้จ่าย
          </span>
        </div>
        <div
          class="border-t border-neutral-200 pt-4"
        >
          <div
            class="flex justify-between"
          >
            <span
              class="text-neutral-600"
            >
              บัญชีปลายทาง
            </span>
            <div
              class="text-right"
            >
              <p
                class="text-neutral-900 font-medium"
              >
                @prayuth789
              </p>
              <p
                class="text-sm text-neutral-600"
              >
                PayWise Account
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="space-y-3"
    >
      <button
        class="w-full bg-web-green-500 hover:bg-web-green-600 text-white py-3 px-4 rounded-lg font-medium transition-colors"
      >
        แสดงใบเสร็จ
      </button>
      <div
        class="grid grid-cols-2 gap-3"
      >
        <button
          class="bg-white border border-neutral-200 hover:border-web-green-200 text-neutral-700 py-3 px-4 rounded-lg font-medium transition-colors"
        >
          แชร์
        </button>
        <button
          class="bg-white border border-neutral-200 hover:border-web-green-200 text-neutral-700 py-3 px-4 rounded-lg font-medium transition-colors"
        >
          ดาวน์โหลด
        </button>
      </div>
      <button
        class="w-full bg-neutral-100 hover:bg-neutral-200 text-neutral-700 py-3 px-4 rounded-lg font-medium transition-colors"
      >
        โอนซ้ำ
      </button>
    </div>
  </div>
</div>
`;

exports[`TransactionDetail Page Snapshot Tests TransactionDetail page with receipt shown should match snapshot 1`] = `
<div
  class="min-h-screen bg-neutral-50"
>
  <div
    class="bg-gradient-to-r from-web-green-600 to-web-green-500 text-white p-4"
  >
    <div
      class="flex items-center space-x-3 mb-4"
    >
      <a
        class="p-2 hover:bg-white/20 rounded-lg transition-colors"
        href="/home"
      >
        <svg
          class="w-5 h-5"
          fill="none"
          stroke="currentColor"
          viewBox="0 0 24 24"
        >
          <path
            d="M15 19l-7-7 7-7"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
      </a>
      <h1
        class="text-xl font-semibold"
      >
        รายละเอียดธุรกรรม
      </h1>
    </div>
  </div>
  <div
    class="p-4 space-y-4"
  >
    <div
      class="bg-white rounded-lg p-6 shadow"
    >
      <div
        class="text-center mb-6"
      >
        <div
          class="w-16 h-16 rounded-full flex items-center justify-center mx-auto mb-4 bg-error/10"
        >
          <svg
            class="w-8 h-8 text-error"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 11l5-5m0 0l5 5m-5-5v12"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
        </div>
        <p
          class="text-3xl font-bold mb-2 text-error"
        >
          -
          ฿
          500
        </p>
        <p
          class="text-lg font-medium text-neutral-900 mb-1"
        >
          โอนเงินให้
           
          สมชาย ใจดี
        </p>
        <p
          class="text-sm text-neutral-600"
        >
          @somchai123
        </p>
        <div
          class="inline-flex items-center px-3 py-1 rounded-full text-sm font-medium mt-3 bg-success/10 text-success"
        >
          สำเร็จ
        </div>
      </div>
    </div>
    <div
      class="bg-white rounded-lg p-6 shadow"
    >
      <h3
        class="text-lg font-semibold text-neutral-900 mb-4"
      >
        รายละเอียด
      </h3>
      <div
        class="space-y-4"
      >
        <div
          class="flex justify-between"
        >
          <span
            class="text-neutral-600"
          >
            หมายเลขอ้างอิง
          </span>
          <span
            class="text-neutral-900 font-medium"
          >
            REF001234567
          </span>
        </div>
        <div
          class="flex justify-between"
        >
          <span
            class="text-neutral-600"
          >
            วันที่และเวลา
          </span>
          <span
            class="text-neutral-900 font-medium"
          >
            19 มิถุนายน 2568 เวลา 14:30:00
          </span>
        </div>
        <div
          class="flex justify-between"
        >
          <span
            class="text-neutral-600"
          >
            บันทึก
          </span>
          <span
            class="text-neutral-900 font-medium"
          >
            ค่าอาหารเที่ยง
          </span>
        </div>
        <div
          class="flex justify-between"
        >
          <span
            class="text-neutral-600"
          >
            จำนวนเงิน
          </span>
          <span
            class="text-neutral-900 font-medium"
          >
            ฿
            500
          </span>
        </div>
        <div
          class="flex justify-between"
        >
          <span
            class="text-neutral-600"
          >
            ค่าธรรมเนียม
          </span>
          <span
            class="font-medium text-green-600"
          >
            ฟรี
          </span>
        </div>
        <div
          class="flex justify-between"
        >
          <span
            class="text-neutral-600"
          >
            หมวดหมู่
          </span>
          <span
            class="text-neutral-900 font-medium"
          >
            อาหาร
          </span>
        </div>
        <div
          class="border-t border-neutral-200 pt-4"
        >
          <div
            class="flex justify-between"
          >
            <span
              class="text-neutral-600"
            >
              บัญชีปลายทาง
            </span>
            <div
              class="text-right"
            >
              <p
                class="text-neutral-900 font-medium"
              >
                @somchai123
              </p>
              <p
                class="text-sm text-neutral-600"
              >
                PayWise Account
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="space-y-3"
    >
      <button
        class="w-full bg-web-green-500 hover:bg-web-green-600 text-white py-3 px-4 rounded-lg font-medium transition-colors"
      >
        แสดงใบเสร็จ
      </button>
      <div
        class="grid grid-cols-2 gap-3"
      >
        <button
          class="bg-white border border-neutral-200 hover:border-web-green-200 text-neutral-700 py-3 px-4 rounded-lg font-medium transition-colors"
        >
          แชร์
        </button>
        <button
          class="bg-white border border-neutral-200 hover:border-web-green-200 text-neutral-700 py-3 px-4 rounded-lg font-medium transition-colors"
        >
          ดาวน์โหลด
        </button>
      </div>
      <button
        class="w-full bg-neutral-100 hover:bg-neutral-200 text-neutral-700 py-3 px-4 rounded-lg font-medium transition-colors"
      >
        โอนซ้ำ
      </button>
    </div>
  </div>
</div>
`;

exports[`TransactionDetail Page Snapshot Tests TransactionDetail page with valid incoming transaction should match snapshot 1`] = `
<div
  class="min-h-screen bg-neutral-50"
>
  <div
    class="bg-gradient-to-r from-web-green-600 to-web-green-500 text-white p-4"
  >
    <div
      class="flex items-center space-x-3 mb-4"
    >
      <a
        class="p-2 hover:bg-white/20 rounded-lg transition-colors"
        href="/home"
      >
        <svg
          class="w-5 h-5"
          fill="none"
          stroke="currentColor"
          viewBox="0 0 24 24"
        >
          <path
            d="M15 19l-7-7 7-7"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
      </a>
      <h1
        class="text-xl font-semibold"
      >
        รายละเอียดธุรกรรม
      </h1>
    </div>
  </div>
  <div
    class="p-4 space-y-4"
  >
    <div
      class="bg-white rounded-lg p-6 shadow"
    >
      <div
        class="text-center mb-6"
      >
        <div
          class="w-16 h-16 rounded-full flex items-center justify-center mx-auto mb-4 bg-success/10"
        >
          <svg
            class="w-8 h-8 text-success"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
          >
            <path
              d="M17 13l-5 5m0 0l-5-5m5 5V6"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
        </div>
        <p
          class="text-3xl font-bold mb-2 text-success"
        >
          +
          ฿
          1,200
        </p>
        <p
          class="text-lg font-medium text-neutral-900 mb-1"
        >
          รับเงินจาก
           
          นิดา สวยงาม
        </p>
        <p
          class="text-sm text-neutral-600"
        >
          @nida456
        </p>
        <div
          class="inline-flex items-center px-3 py-1 rounded-full text-sm font-medium mt-3 bg-success/10 text-success"
        >
          สำเร็จ
        </div>
      </div>
    </div>
    <div
      class="bg-white rounded-lg p-6 shadow"
    >
      <h3
        class="text-lg font-semibold text-neutral-900 mb-4"
      >
        รายละเอียด
      </h3>
      <div
        class="space-y-4"
      >
        <div
          class="flex justify-between"
        >
          <span
            class="text-neutral-600"
          >
            หมายเลขอ้างอิง
          </span>
          <span
            class="text-neutral-900 font-medium"
          >
            REF001234568
          </span>
        </div>
        <div
          class="flex justify-between"
        >
          <span
            class="text-neutral-600"
          >
            วันที่และเวลา
          </span>
          <span
            class="text-neutral-900 font-medium"
          >
            18 มิถุนายน 2568 เวลา 19:45:00
          </span>
        </div>
        <div
          class="flex justify-between"
        >
          <span
            class="text-neutral-600"
          >
            บันทึก
          </span>
          <span
            class="text-neutral-900 font-medium"
          >
            คืนเงินค่าหนัง
          </span>
        </div>
        <div
          class="flex justify-between"
        >
          <span
            class="text-neutral-600"
          >
            จำนวนเงิน
          </span>
          <span
            class="text-neutral-900 font-medium"
          >
            ฿
            1,200
          </span>
        </div>
        <div
          class="flex justify-between"
        >
          <span
            class="text-neutral-600"
          >
            ค่าธรรมเนียม
          </span>
          <span
            class="font-medium text-green-600"
          >
            ฟรี
          </span>
        </div>
        <div
          class="flex justify-between"
        >
          <span
            class="text-neutral-600"
          >
            หมวดหมู่
          </span>
          <span
            class="text-neutral-900 font-medium"
          >
            บันเทิง
          </span>
        </div>
        <div
          class="border-t border-neutral-200 pt-4"
        >
          <div
            class="flex justify-between"
          >
            <span
              class="text-neutral-600"
            >
              บัญชีปลายทาง
            </span>
            <div
              class="text-right"
            >
              <p
                class="text-neutral-900 font-medium"
              >
                @nida456
              </p>
              <p
                class="text-sm text-neutral-600"
              >
                PayWise Account
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="space-y-3"
    >
      <button
        class="w-full bg-web-green-500 hover:bg-web-green-600 text-white py-3 px-4 rounded-lg font-medium transition-colors"
      >
        แสดงใบเสร็จ
      </button>
      <div
        class="grid grid-cols-2 gap-3"
      >
        <button
          class="bg-white border border-neutral-200 hover:border-web-green-200 text-neutral-700 py-3 px-4 rounded-lg font-medium transition-colors"
        >
          แชร์
        </button>
        <button
          class="bg-white border border-neutral-200 hover:border-web-green-200 text-neutral-700 py-3 px-4 rounded-lg font-medium transition-colors"
        >
          ดาวน์โหลด
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`TransactionDetail Page Snapshot Tests TransactionDetail page with valid outgoing transaction should match snapshot 1`] = `
<div
  class="min-h-screen bg-neutral-50"
>
  <div
    class="bg-gradient-to-r from-web-green-600 to-web-green-500 text-white p-4"
  >
    <div
      class="flex items-center space-x-3 mb-4"
    >
      <a
        class="p-2 hover:bg-white/20 rounded-lg transition-colors"
        href="/home"
      >
        <svg
          class="w-5 h-5"
          fill="none"
          stroke="currentColor"
          viewBox="0 0 24 24"
        >
          <path
            d="M15 19l-7-7 7-7"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
      </a>
      <h1
        class="text-xl font-semibold"
      >
        รายละเอียดธุรกรรม
      </h1>
    </div>
  </div>
  <div
    class="p-4 space-y-4"
  >
    <div
      class="bg-white rounded-lg p-6 shadow"
    >
      <div
        class="text-center mb-6"
      >
        <div
          class="w-16 h-16 rounded-full flex items-center justify-center mx-auto mb-4 bg-error/10"
        >
          <svg
            class="w-8 h-8 text-error"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 11l5-5m0 0l5 5m-5-5v12"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
        </div>
        <p
          class="text-3xl font-bold mb-2 text-error"
        >
          -
          ฿
          500
        </p>
        <p
          class="text-lg font-medium text-neutral-900 mb-1"
        >
          โอนเงินให้
           
          สมชาย ใจดี
        </p>
        <p
          class="text-sm text-neutral-600"
        >
          @somchai123
        </p>
        <div
          class="inline-flex items-center px-3 py-1 rounded-full text-sm font-medium mt-3 bg-success/10 text-success"
        >
          สำเร็จ
        </div>
      </div>
    </div>
    <div
      class="bg-white rounded-lg p-6 shadow"
    >
      <h3
        class="text-lg font-semibold text-neutral-900 mb-4"
      >
        รายละเอียด
      </h3>
      <div
        class="space-y-4"
      >
        <div
          class="flex justify-between"
        >
          <span
            class="text-neutral-600"
          >
            หมายเลขอ้างอิง
          </span>
          <span
            class="text-neutral-900 font-medium"
          >
            REF001234567
          </span>
        </div>
        <div
          class="flex justify-between"
        >
          <span
            class="text-neutral-600"
          >
            วันที่และเวลา
          </span>
          <span
            class="text-neutral-900 font-medium"
          >
            19 มิถุนายน 2568 เวลา 14:30:00
          </span>
        </div>
        <div
          class="flex justify-between"
        >
          <span
            class="text-neutral-600"
          >
            บันทึก
          </span>
          <span
            class="text-neutral-900 font-medium"
          >
            ค่าอาหารเที่ยง
          </span>
        </div>
        <div
          class="flex justify-between"
        >
          <span
            class="text-neutral-600"
          >
            จำนวนเงิน
          </span>
          <span
            class="text-neutral-900 font-medium"
          >
            ฿
            500
          </span>
        </div>
        <div
          class="flex justify-between"
        >
          <span
            class="text-neutral-600"
          >
            ค่าธรรมเนียม
          </span>
          <span
            class="font-medium text-green-600"
          >
            ฟรี
          </span>
        </div>
        <div
          class="flex justify-between"
        >
          <span
            class="text-neutral-600"
          >
            หมวดหมู่
          </span>
          <span
            class="text-neutral-900 font-medium"
          >
            อาหาร
          </span>
        </div>
        <div
          class="border-t border-neutral-200 pt-4"
        >
          <div
            class="flex justify-between"
          >
            <span
              class="text-neutral-600"
            >
              บัญชีปลายทาง
            </span>
            <div
              class="text-right"
            >
              <p
                class="text-neutral-900 font-medium"
              >
                @somchai123
              </p>
              <p
                class="text-sm text-neutral-600"
              >
                PayWise Account
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="space-y-3"
    >
      <button
        class="w-full bg-web-green-500 hover:bg-web-green-600 text-white py-3 px-4 rounded-lg font-medium transition-colors"
      >
        แสดงใบเสร็จ
      </button>
      <div
        class="grid grid-cols-2 gap-3"
      >
        <button
          class="bg-white border border-neutral-200 hover:border-web-green-200 text-neutral-700 py-3 px-4 rounded-lg font-medium transition-colors"
        >
          แชร์
        </button>
        <button
          class="bg-white border border-neutral-200 hover:border-web-green-200 text-neutral-700 py-3 px-4 rounded-lg font-medium transition-colors"
        >
          ดาวน์โหลด
        </button>
      </div>
      <button
        class="w-full bg-neutral-100 hover:bg-neutral-200 text-neutral-700 py-3 px-4 rounded-lg font-medium transition-colors"
      >
        โอนซ้ำ
      </button>
    </div>
  </div>
</div>
`;
